cpp/src/sort/top_k.cu

```diff
@@ -37,9 +38,10 @@ std::unique_ptr<column> top_k(column_view const& col,
                               rmm::cuda_stream_view stream,
                               rmm::device_async_resource_ref mr)
 {
-  CUDF_EXPECTS(k <= col.size(),
-               "k must be less than or equal to the number of rows in the column",
+  CUDF_EXPECTS(k >= 0 && k <= col.size(),
+               "k must be non-negative and less than or equal to the number of input rows",
```

Comment:
The restriction for `<= col.size()` might be unnecessary enforcement. Is that required for correctness? I would expect from the docstrings that it could just return all the items of each segment in any order.

I am imagining a use case where users know they want a maximum of 100 items per segment, and thus provide `k=100`, but in specific edge cases the dataset might be only a few rows because it comes from chunking a larger dataset.

Reply:
Ok, that makes sense.